src/vde_device.c

```diff
@@ -102,7 +101,7 @@ static void close_device(void) {
 }
 
 static bool read_packet(vpn_packet_t *packet) {
-	int lenin = (ssize_t)plug.vde_recv(conn, DATA(packet), MTU, 0);
+	ssize_t lenin = (ssize_t) plug.vde_recv(conn, DATA(packet), MTU, 0);
```

Comment:
Hm, I don't know why the explicit cast was necessary... according to the manpage, `vde_recv()` already returns a `ssize_t`...

Reply:
This was triggered by testing on one of the supported platforms (probably FreeBSD). The latest version does indeed declare the returned type as `ssize_t`:

https://github.com/bringhurst/vde/blob/master/include/libvdeplug_dyn.h#L89

However, the same header on my desktop OS has `size_t` there. FreeBSD 13 has `size_t`, too.

They wiped their change history when moving from svn to git, so here's that change in someone else's fork of the project:

https://github.com/bringhurst/vde/commit/385d4b49b05c4b8b2ecdc695ec42a863024e7da6#diff-8744b91559c96b744dd3f015a4cd367daccb11a92219e9e4f4cb2d501cf36a30